crates/fluvio-spu/src/replication/test.rs

```diff
@@ -861,3 +871,82 @@ async fn test_sync_2_replicas_but_one_reject() {
 
     spu_server.notify();
 }
+
+#[fluvio_future::test(ignore)]
+async fn test_sync_larger_records() {
+    let num_records = 5;
+    let port = portpicker::pick_unused_port().expect("No free ports left");
+    let builder = TestConfig::builder()
+        .followers(1_u16)
+        .base_port(port)
+        .generate("sync_larger_records");
+
+    let leader_gctx = builder.leader_ctx().await;
+
+    let spu_server = create_internal_server(builder.leader_addr(), leader_gctx.clone()).run();
+
+    let replica = builder.replica();
+
+    let actions = leader_gctx
+        .apply_replica_update(UpdateReplicaRequest::with_all(1, vec![replica.clone()]))
+        .await;
+    assert!(actions.is_empty());
+
+    // give leader controller time to startup
+    sleep(Duration::from_millis(MAX_WAIT_LEADER)).await;
+
+    let leader = leader_gctx
+        .leaders_state()
+        .get(&replica.id)
+        .await
+        .expect("replica");
+    assert!(leader_gctx
+        .followers_state()
+        .get(&replica.id)
+        .await
+        .is_none());
+    // should be new
+    assert_eq!(leader.leo(), 0);
+    assert_eq!(leader.hw(), 0);
+
+    // create a raw record with 512kb
+    let record_test: Vec<u8> = vec![10; 512 * 1024];
```

Comment:
create unique records instead of same to validate that records have all arrived intact

Reply:
done